api/service/query_type.go

```diff
@@ -0,0 +1,60 @@
+package service
+
+import (
+	"regexp"
+
+	"github.com/gin-gonic/gin"
+
+	"github.com/photon-storage/go-photon/crypto/bls"
+
+	"github.com/photon-storage/photon-explorer/database/orm"
+)
+
+type queryResp struct {
+	Type string `json:"type"`
+}
+
+var (
+	slotRegMatch      = `^\d+$`
+	hashRegMatch      = "^[0-9a-fA-F]{64}$"
+	publicKeyRegMatch = "^[0-9a-fA-F]{96}$"
+)
+
+// QueryType handles the /query request.
+func (s *Service) QueryType(c *gin.Context) (*queryResp, error) {
+	value := c.Query("value")
+	if value == "" {
+		return nil, errMissingQueryValue
+	}
+
+	queryType := "block"
+	switch {
+	case regexp.MustCompile(slotRegMatch).MatchString(value):
```

Comment:
why not compile once in constructor? There is no need to compile the same pattern every time?